### DISCUSSION TITLE
How to use KeyDown event in CanvasControl?

### DISCUSSION THREAD

**Discussion by HASPzhu** [microsoft/microsoft-ui-xaml#7997] [Q&A] [answered]
I want to use KeyDown event and it did't work. 
I checked it with Microsoft.UI.Xaml.Input.FocusManager.GetFocusedElement(ContentArea.XamlRoot).GetType().Name. It is ScrollViewer when I click on CanvasControl area. /*ContentArea is a grid in page*/
what shall I do to resolve it?

**Comment by castorix** [comment, marked as answer]
A way to read input with focus on any control is with Hooks
You can see this thread where I had posted a test sample : [Gathering keyboard input doesn't work (winui 3) #7330](https://github.com/microsoft/microsoft-ui-xaml/issues/7330)

**Comment by nikolayvpavlov** [comment]
Set the Backgound of the canvas to _Transparent_: 
`Background="Transparent"`
